pandas/core/generic.py

```diff
@@ -5601,53 +5601,98 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
         """
         Trim values at input threshold(s).
 
+        Elements above/below the upper/lower thresholds will be changed to
+        upper/lower thresholds.
+
         Parameters
         ----------
-        lower : float or array_like, default None
-        upper : float or array_like, default None
-        axis : int or string axis name, optional
+        lower : float, array-like or None, default None
+            Lower threshold for clipping.  Values smaller than `lower` will be
+            converted to `lower`.
+        upper : float, array-like or None, default None
+            Upper threshold for clipping.  Values larger than `upper` will be
+            converted to `upper`.
+        axis : {0 or 'index', 1 or 'columns', None}, default None
             Align object with lower and upper along the given axis.
         inplace : boolean, default False
             Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+                .. versionadded:: 0.21.0.
+        *args : arguments passed to pandas.compat.numpy
+        **kwargs : keyword arguments passed to pandas.compat.numpy
 
         Returns
         -------
-        clipped : Series
+        clipped : `Series` or `DataFrame`.
+            Elements above or below the upper and lower thresholds converted to
+            threshold values.
```

Comment:
We can just have the type in the first row of Returns, providing a name doesn't add much value.

The description sounds a bit like if we could be returning only part of the original values.

Reply:
Sure thing.